### PR TITLE
Location Checkin, Event population with script, Reformatted Datetime objects, etc.

### PR DESCRIPTION
**Pull request by Saisriram2003**

### CODE REVIEW

File: bu_passport/lib/pages/login_page.dart
Comment (molly-yan) [inline bu_passport/lib/pages/login_page.dart:74]: - Show error message on frontend, instead of as a log message

File: bu_passport/lib/pages/onboarding_page.dart
Comment (molly-yan) [inline bu_passport/lib/pages/onboarding_page.dart]: Use dynamic measurements, instead of magic numbers

File: bu_passport/lib/pages/event_page.dart
Comment (molly-yan) [inline bu_passport/lib/pages/event_page.dart]: - Is "location too far" the only false case here, or there might be other possible false cases, such as "Location permission not granted"?

File: bu_passport/lib/services/firebase_service.dart
Comment (molly-yan) [inline bu_passport/lib/services/firebase_service.dart]: Delete unused code

File: bu_passport/lib/services/geocoding_service.dart
Comment (molly-yan) [inline bu_passport/lib/services/geocoding_service.dart]: Need to include the catch error case

File: bu_passport/scripts/eventscraper.py
Comment (molly-yan) [inline bu_passport/scripts/eventscraper.py:87]: It's fine for now, but since we might consider using the ICS file provided by IS&T, this method might need to be replaced later.

**Review by molly-yan — CHANGES_REQUESTED** [review]
Good work! Please view the comments for things that need adjustments. 

As a reminder, please be careful with committing config files for future development, which might include identifier id/id strings that are unique to your XCode runner.

File: bu_passport/ios/.symlinks/plugins/cloud_firestore
Comment (molly-yan) [inline bu_passport/ios/.symlinks/plugins/cloud_firestore:1]: Please prevent committing config files for future PRs.

File: bu_passport/lib/pages/onboarding_page.dart
Comment (lcfmarco) [reply bu_passport/lib/pages/onboarding_page.dart]: Just made the change + some additional margins for the onboarding page. I think we are good to merge after your approval? The logo images have this weird blue line across it but I think Zihan can upload new logo photos to remove that after

**Review by molly-yan — APPROVED** [review]
Good to be merged.

**Pull request merged**
